tests/integration_tests/functional/test_cpu_features.py

```diff
@@ -174,19 +198,26 @@ def test_cpu_template(test_microvm_with_ssh, network_config, cpu_template):
     assert test_microvm.api_session.is_status_no_content(
             response.status_code)
 
+    # Check that all common features discoverable with lscpu
+    # are properly masked.
     ssh_connection = net_tools.SSHConnection(test_microvm.ssh_config)
     guest_cmd = "cat /proc/cpuinfo | grep 'flags' | head -1"
     _, stdout, stderr = ssh_connection.execute_command(guest_cmd)
     assert stderr.read() == ''
 
-    cpu_flags_output = stdout.readline().rstrip()
+    cpu_flags_output = stdout.readline().rstrip().split(' ')
+
+    for feature in common_masked_features_lscpu:
+        assert feature not in cpu_flags_output, feature
+
+    # Check that all common features discoverable with cpuid
+    # are properly masked.
+    utils.check_guest_cpuid_output(test_microvm, "cpuid -1", None, '=',
```

Comment:
Do we need to check for both `/proc/cpuinfo` and `cpuid -1` or is just for extra safety?

Reply:
The reason we use both `lscpu` and `cpuid -1` is that not all masked features are detected by `lscpu`. And also, as far as my observation, there are some features that `cpuid` discovers but `lscpu` does not. I tried to make sure that each feature we masked are either discoverable through `lscpu` or `cpuid -1`. You can see that the set of features from `common_masked_features_lscpu ` is different than that of `common_masked_features_cpuid `.

Reply:
Indeed. Thanks. Resolving this.